client/src/AppContext.js

```diff
@@ -1,22 +1,100 @@
 import React, { Component } from 'react';
+import axios from 'axios';
 
+//const URL = process.env.REACT_APP_LOCAL_BACKEND_URL || 'http://127.0.0.1:8000/api';
+const URL =
+  process.env.REACT_APP_REMOTE_BACKEND_URL ||
+  'https://scratch-n-map.herokuapp.com/api';
+
+const DEV = process.env.REACT_APP_DEV || true;
+
+// initialize the React Context API
 const AppContext = React.createContext();
 
 export class AppContextProvider extends Component {
-	state = { authenticated: false };
-
-	render() {
-		return (
-			<AppContext.Provider
-				value={{
-					AppState: this.state,
-					authenticated: this.state.authenticated,
-				}}
-			>
-				{this.props.children}
-			</AppContext.Provider>
-		);
-	}
+  state = {
+    authenticated: false,
+    error: ''
+  };
+  componentDidMount() {
+    console.log('THE begining');
+  }
+  login = user => {
+    (async () => {
+      try {
+        const response = await axios.post(`${URL}/login`, user);
+
+        const data = response.data;
+
+        if (data.jwt_token) {
+          // set the token to local storage
+          localStorage.setItem('jwt_token', data.jwt_token);
+
+          // reset the fields
+          this.setState({
+            authenticated: true,
+            error: ''
+          });
+          window.location.reload();
```

Comment:
Forcing a page refresh is causing the AppState to be reset, which is an issue. We'll need to persist some user data on AppState so it can be used for subsequent API calls.

Reply:
The get user is not in this branch, I have to fix the force reload before putting a PR. I have another branch I was working on.